### PR TITLE
[CM-BE-6] | Leela Prasanth | Refactors the code

### PR DESCRIPTION
**Pull request by leelanakka**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/LockedUserService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5517327&bid=18078096)

File: src/main/java/in/projecteka/consentmanager/user/SessionService.java [hunk 1]
```diff
@@ -49,24 +50,19 @@
...
+        var newLockedUser = new LockedUser(0, request.getUsername(), false, "", new Date().toString());
 
-        return lockedUser
+        return lockedUserService.userFor(request.getUsername())
+                .switchIfEmpty(
+                        Mono.just(newLockedUser))
                 .flatMap(user -> {
                     if (lockedUserService.isUserBlocked(user)) {
-                        return Mono.error(ClientError.userBlocked());
+                        return lockedUserService.validateAndUpdate(user).then(Mono.error(ClientError.userBlocked()));
```
Comment (Apoorva-GA) [inline src/main/java/in/projecteka/consentmanager/user/SessionService.java:60]: If the user is Blocked, why are we validating again? Also what are we updating?

File: src/main/java/in/projecteka/consentmanager/user/LockedUserService.java [hunk 2]
```diff
@@ -38,6 +38,10 @@ public boolean isUserBlocked(LockedUser user) {
         var blockedTime = user.getLockedTime();
         var firstInvalidAttempt = user.getFirstInvalidAttemptTime();
 
+        if (user.getInvalidAttempts() == 0) {
+            return createUser(user.getPatientId())
+                    .then(Mono.error(ClientError.unAuthorizedRequest("Username or password is incorrect")));
```
[the remark this answers is not in the document]
Comment (leelanakka) [reply src/main/java/in/projecteka/consentmanager/user/LockedUserService.java:43]: Sure, will do that

File: src/main/java/in/projecteka/consentmanager/user/SessionService.java [hunk 3]
```diff
@@ -49,24 +50,19 @@
...
-        var newLockedUser = new LockedUser(0, request.getUsername(), false, "", "");
-        Mono<LockedUser> lockedUser = lockedUserService.userFor(request.getUsername())
-                .switchIfEmpty(
-                        Mono.just(newLockedUser));
+        var newLockedUser = new LockedUser(0, request.getUsername(), false, "", new Date().toString());
 
-        return lockedUser
+        return lockedUserService.userFor(request.getUsername())
+                .switchIfEmpty(
+                        Mono.just(newLockedUser))
```
[the remark this answers is not in the document]
Comment (leelanakka) [reply src/main/java/in/projecteka/consentmanager/user/SessionService.java:57]: sure

File: src/main/java/in/projecteka/consentmanager/user/SessionService.java (hunk 1 above)
Comment (leelanakka) [reply src/main/java/in/projecteka/consentmanager/user/SessionService.java:60]: we are updating the invalid_attempts

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/LockedUserService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5517327&bid=18078096)

**Comment by Apoorva-GA** [comment]
Please fix the code smells and merge.

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/in/projecteka/consentmanager/user/LockedUserService.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/consent-manager/pullRequest?prid=5517327&bid=18078096)

**Pull request merged**